core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/upgrade/DistroXUpgradeAvailabilityService.java

```diff
@@ -91,6 +95,33 @@ public UpgradeV4Response checkForUpgrade(NameOrCrn nameOrCrn, Long workspaceId,
         return filteredCandidates;
     }
 
+    private void filterOnlyPatchUpgradesIfRuntimeUpgradeDisabled(String accountId, String clusterName, UpgradeV4Response upgradeV4Response) {
+        if (!isRuntimeUpgradeEnabledByAccountId(accountId)) {
+            if (upgradeV4Response.getCurrent() != null) {
+                List<ImageInfoV4Response> upgradeCandidates = upgradeV4Response.getUpgradeCandidates();
+                String currentCdpVersion = upgradeV4Response.getCurrent().getComponentVersions().getCdp();
+                LOGGER.debug("Only patch upgrade is possible on [{}] cluster for [{}] runtime as CDP_RUNTIME_UPGRADE is disabled in [{}] account.",
+                        clusterName, currentCdpVersion, accountId);
+                upgradeCandidates = upgradeCandidates.stream()
+                        .filter(upgradeCandidate -> upgradeCandidate.getComponentVersions().getCdp().equals(currentCdpVersion))
+                        .collect(Collectors.toList());
+                upgradeV4Response.setUpgradeCandidates(upgradeCandidates);
+                if (upgradeCandidates.isEmpty()) {
+                    upgradeV4Response.setReason("No image is available for maintenance upgrade");
```

Comment:
minor: we might indicate the cdp version in the reason to be matched for extra help for the user

Reply:
added